src/utils.ts

```diff
@@ -277,6 +277,19 @@ export function convertByteNumToCharNum(line: string, col: number): number {
     return currCharNum;
 }
 
+export function convertByteNumToCharNumOnTop(line: string, col: number): number {
+    let totalBytes = 0;
+    let currCharNum = 0;
+    while (totalBytes < col) {
+        if (currCharNum >= line.length) {
+            return currCharNum + (col - totalBytes);
```

Comment:
Why this is needed? Anyway, better to modify the existing function rather than creating a copy with slighly changed behavior